src/base/area_alloc.c

```diff
@@ -149,14 +160,21 @@ area_create (const char *name, size_t element_size, size_t alloc_count)
   size_t adjust;
 #if defined (SERVER_MODE)
   int rv;
+  MMON_STAT_ID prev_tag = mmon_set_tracking_tag (MMON_COMMON);
```

Comment:
You can make `(void) mmon_set_tracking_tag (prev_tag);` called automatically and safely with an RAII object. It doesn't only look good, but it's also safe. Please consider this. 
Example:
```
class mmon_tag_tracker
{
        public:
        mmon_tag_tracker (int tag)
        {
                cur_tag = tag;
                prev_tag = mmon_set_tracking_tag (tag);
        }

        ~mmon_tag_tracker ()
        {
                int ttt =  mmon_set_tracking_tag (prev_tag);
                assert (origin_tag == ttt);
        }

        private:
                int prev_tag;
                int origin_tag; 
}

```
```suggestion
  mmon_tag_tracker tracker track (MMON_COMMON);
```

Reply:
It can be added c-only code.

Reply:
A c-complier can compile this code, can't it? It's just a declaration of a structure in the respective of c-compiler.